pkg/vulnsrc/amazon/amazon.go

```diff
@@ -0,0 +1,184 @@
+		}
+
+		var vuln amazon.ALAS
+		if err = json.NewDecoder(r).Decode(&vuln); err != nil {
+			return xerrors.Errorf("failed to decode amazon JSON: %w", err)
+		}
+
+		alasList = append(alasList, alas{
+			Version: version,
+			ALAS:    vuln,
+		})
+		bar.Increment()
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in amazon walk: %w", err)
+	}
+
+	if err = ac.save(alasList); err != nil {
+		return xerrors.Errorf("error in amazon save: %w", err)
+	}
+
+	return nil
+}
+
+func (ac Config) save(alasList []alas) error {
+	ac.lg.Debug("Saving amazon DB")
+	err := ac.dbc.BatchUpdate(func(tx *bolt.Tx) error { // TODO: Untested
+		for _, alas := range alasList {
+			for _, cveID := range alas.CveIDs {
+				for _, pkg := range alas.Packages {
+					platformName := fmt.Sprintf(platformFormat, alas.Version)
+					advisory := vulnerability.Advisory{
+						VulnerabilityID: cveID,
+						FixedVersion:    constructVersion(pkg.Epoch, pkg.Version, pkg.Release),
+					}
+					if err := ac.dbc.PutNestedBucket(tx, platformName, pkg.Name, cveID, advisory); err != nil {
+						return xerrors.Errorf("failed to save amazon advisory: %w", err)
+					}
+
+					var references []string
+					for _, ref := range alas.References {
+						references = append(references, ref.Href)
+					}
+
+					vuln := vulnerability.Vulnerability{
+						Severity:    severityFromPriority(alas.Severity),
+						References:  references,
+						Description: alas.Description,
+						// TODO
+						Title: "",
+					}
+					if err := vulnerability.Put(tx, cveID, vulnerability.Amazon, vuln); err != nil {
+						return xerrors.Errorf("failed to save amazon vulnerability: %w", err)
+					}
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in batch update: %w", err)
+	}
+	return nil
+}
+
+// Get returns a security advisory
+func (ac Config) Get(version string, pkgName string) ([]vulnerability.Advisory, error) {
+	bucket := fmt.Sprintf(platformFormat, version)
+	advisories, err := ac.dbc.ForEach(bucket, pkgName)
+	if err != nil {
+		return nil, xerrors.Errorf("error in amazon foreach: %w", err)
+	}
+	if len(advisories) == 0 {
+		return nil, nil
+	}
+
+	var results []vulnerability.Advisory
+	for _, v := range advisories {
+		var advisory vulnerability.Advisory
+		if err = json.Unmarshal(v, &advisory); err != nil {
+			return nil, xerrors.Errorf("failed to unmarshal amazon JSON: %w", err)
+		}
+		results = append(results, advisory)
+	}
+	return results, nil
+}
+
+func severityFromPriority(priority string) vulnerability.Severity {
+	switch priority {
+	case "low":
+		return vulnerability.SeverityLow
+	case "medium":
+		return vulnerability.SeverityMedium
+	case "important":
+		return vulnerability.SeverityHigh
+	case "critical":
+		return vulnerability.SeverityCritical
+	default:
+		return vulnerability.SeverityUnknown
+	}
+}
+
+func constructVersion(epoch, version, release string) string {
+	verStr := ""
+	if epoch != "0" && epoch != "" {
```

Comment:
isn't this essentially equivalent to `if epoch != "0"`?

Reply:
`epoch` is sometimes empty. In that case, a version becomes `:1.2.3`. This should be `1.2.3`.